lib/iiif_print/metadata.rb

```diff
@@ -0,0 +1,116 @@
+module IiifPrint
+  # rubocop:disable Metrics/ModuleLength, Metrics/ClassLength
+  class Metadata
+    def self.manifest_for(model:, version:, fields:)
+      new(model: model, version: version, fields: fields).manifest
+    end
+
+    def initialize(model:, version:, fields:)
+      @model = model
+      @version = version
+      @fields = fields
+    end
+
+    attr_reader :model, :version, :fields
+
+    # rubocop:disable MethodLength, CyclomaticComplexity, PerceivedComplexity, AbcSize
+    def manifest
+      case version
+      when 2
+        fields.map do |field|
+          options = field.options
+          parent = model.try(:model)
+          next if parent
+          label = Hyrax::Renderers::AttributeRenderer.new(field.name, nil).label
+          if field.name == :collection
+            next unless model[:member_of_collection_ids_ssim]&.present?
+            viewable_collections = Hyrax::CollectionMemberService.run(model, @current_ability)
+            next if viewable_collections.blank?
+            {
+              'label' => label,
+              'value' => make_collection_link(viewable_collections)
+            }
+          else
+            next if Array(model.try(field.name)).first.blank?
```

Comment:
I want to talk through this with you.  You making a complex method call here, then likely inside `cast_to_value` and also in the v3 look.

Which feels like you've got a method that is begging to exist.  Because `Array(model.try(field.name))` is already ugly enough.